### PR TITLE
DTFS2-5406 - TFM - Added facility security details on deals page

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Changes made: 

* Added facility security heading and text on TFM
* Only display if exists (does not exist on BSS)
* Added to e2e tests to test works
* Added to deals page component test

### CODE REVIEW

File: trade-finance-manager-ui/templates/case/deal/deal.njk [hunk 1]
```diff
@@ -53,9 +53,16 @@
...
 
-      <p style="white-space:pre-line" data-cy="bank-security-text">{{ deal.supportingInformation.securityDetails.exporter }}</p>
+        <p style="white-space:pre-line" data-cy="bank-security-text">{{ deal.supportingInformation.securityDetails.exporter }}</p>
+        {% if deal.supportingInformation.securityDetails.facility %}
+          <h4 class="govuk-heading-s govuk-!-padding-top-3 govuk-!-padding-bottom-0 govuk-!-margin-top-0" data-cy="bank-security-facility-sub-heading">
+          Specific bank security for the facilities listed in this application
+          </h4>
+
+          <p style="white-space:pre-line" data-cy="bank-security-facility-text">{{ deal.supportingInformation.securityDetails.facility }}</p>
+        {% endif %}
```
Comment (ttbarnes) [inline trade-finance-manager-ui/templates/case/deal/deal.njk:65]: In the PR it says this only exists for GEF - should we be adding a check here for the dealType?

File: e2e-tests/trade-finance-manager/cypress/integration/journeys/case/deal-MIA.spec.js [hunk 2]
```diff
@@ -64,6 +64,8 @@ context('User can view a case deal', () => {
       pages.caseDealPage.bankSecuritySectionHeading().contains('Bank security');
       pages.caseDealPage.bankSecuritySubHeading().contains('General bank security for this exporter');
       pages.caseDealPage.bankSecurityText().contains('Mock security details');
+      pages.caseDealPage.bankSecurityFacilitySubHeading().contains('Specific bank security for the facilities listed in this application');
+      pages.caseDealPage.bankSecurityFacilityText().contains('Mock facility details');
```
Comment (ttbarnes) [inline e2e-tests/trade-finance-manager/cypress/integration/journeys/case/deal-MIA.spec.js:68]: If it should only display/exist for GEF deal this test is misleading as it's using a mock BSS deal

**Review by ttbarnes — CHANGES_REQUESTED** [review]

File: trade-finance-manager-ui/templates/case/deal/deal.njk (hunk 1 above)
Comment (Zainzzkk) [reply trade-finance-manager-ui/templates/case/deal/deal.njk:65]: the check just above :P BSS won't have deal.supportingInformation.securityDetails.facility so the if statement will stop it showing

File: e2e-tests/trade-finance-manager/cypress/support/portal-api/createGefFacility.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        facilityWithId,
+        facilityWithId.type,
+        user,
+        token,
+      ).then((createdFacilities) => {
+        const facilityId = createdFacilities.details._id;
+        updateGefFacilities(facilityId, facilityToInsert, token).then((updated) => updated);
+      });
+    });
+  //  return Promise.resolve(facilitiesWithUkefIds);
```
Comment (ttbarnes) [inline e2e-tests/trade-finance-manager/cypress/support/portal-api/createGefFacility.js:26]: can remove this

**Review by ttbarnes — APPROVED** [review]
LGTM minor comment  not a blocker

**Pull request merged**